src/plugins/intel_gpu/src/graph/impls/onednn/primitive_onednn_base.h

```diff
@@ -490,16 +494,23 @@ struct typed_primitive_onednn_impl : public typed_primitive_impl<PType> {
                 throw;    // rethrowing dnnl::error if not out_of_memory
             }
 
-            // If oneDNN primitive is the output primitive or it's user is CPU implementation, then enqueue marker
-            // with empty events wait list (which will trigger wait for all previously enqueued tasks) and
-            // return it as oneDNN primitive's event as it is a single option for proper synchronization
-            if (instance.needs_completion_event())
-                event = stream.enqueue_marker({});
-        }
+            if (_enable_profiling) {
+                // Call wait() function here instead of finish() to prevent cache flushing,
+                // this synchronization point is needed for correct OneDNN's profiling process
+                stream.wait();
```

Comment:
@isanghao, yeah, good idea, don't you mind to implement it in a separate PR? Just because it requires a little more tests to make sure oneDNN will handle it properly